protobuf_definitions/message_formats.proto

```diff
@@ -363,6 +363,45 @@ message PositionEstimate {
   repeated NavigationSensorStatus navigation_sensors = 11; // List of available sensors with status
 }
 
+/*
+ * Waypoint used to describe a location and depth on a map.
+ */
+message Waypoint {
+  uint32 id = 1; // waypoint id
+  LatLongPosition global_position =2; // position if the waypoint (decimal degrees)
+  float circle_of_acceptance = 3; // radius of the accepance circle around the waypoint (m)
+  float speed_over_ground = 4; // desired speed to waypoint (m/s)
+  float depth = 5; // desired depth at the wp (m)
+  float depth_speed = 6; // desired speed to desired depth set point (m/s)
+  DepthZeroReference depth_zero_reference = 7; // used to destinguish desired altitude or depth
+  bool visited = 8; // indicates if the point is visited during current mission
+  string name = 9; // waypoint name provided from the app
+}
+
+/*
+ * A list of waypoints describes a mission that the auto pilot can execute.
+ */
+message Mission {
+  uint32 id = 1; // mission id
+  repeated Waypoint waypoints = 2; // list of waypoints in the mission
+}
+
+/*
+ * Path segment used to describe segments of a mission as a line between to waypoints.
+ */
+message PathSegment {
+  uint32 id = 1; // path segment id starting at 0, -1 for inactive
+  float speed_over_ground = 2; // desired speed over ground in (m/s)
+  float course_to_target = 3; // course to target relative to north (radians)
+  float depth_speed = 4; // desired speed in heave (m/s)
+  float horizontal_length = 5; // horizontal length of the path segment (m)
+  float vertical_length = 6; // vertical legth of the path segment (m)
+  bool completed = 7; // indicates if the path segment is completed in the mission
+  uint32 from_wp_id = 8; // id of the starting waypoint
```

Comment:
Maybe use the waypoint reference directly? If you need to get the id, you can always use `from_wp.id`